src/run/run.c

```diff
@@ -1045,6 +1058,25 @@ static int parse_argv_sudo_mode(int argc, char *argv[]) {
                         return log_error_errno(r, "Failed to set $SHELL_PROMPT_PREFIX environment variable: %m");
         }
 
+        /* When using run0 to acquire privileges temporarily, let's not pull in session manager by
+         * default. Note that pam_logind/systemd-logind doesn't distinguish between run0-style privilege
+         * escalation and first class TTY logins (and thus gives root a per-session manager for interactive
+         * TTY sessions), hence let's override the logic explicitly here. */
+        if (arg_lightweight < 0 && !strv_env_get(arg_environment, "XDG_SESSION_CLASS") && privileged_execution())
```

Comment:
I'd like to add a comment that we only mangle class at client if the target is rootz otherwise it's delegated to logind.